StardewArchipelago/Archipelago/Gifting/CrossGiftHandler.cs

```diff
@@ -47,8 +48,8 @@ public void Initialize(IMonitor monitor, ArchipelagoClient archipelago, StardewI
             _mail = mail;
             _archipelago = archipelago;
             _giftService = new GiftingService(archipelago.Session);
-            _giftSender = new GiftSender(_monitor, _archipelago, _itemManager, _giftService);
-            _giftReceiver = new GiftReceiver(_monitor, _archipelago, _giftService, _itemManager, _mail);
+            _giftSender = new GiftSender(_monitor, _archipelago, _itemManager, _giftService, this);

```

Comment:
Don't pass "this", that's a circular dependency. Pass the collection of IDs instead and let both of these classes handle it on their side. If made a hashset, it'll be very simple